explorer/testdata/choice/generic_choice1.carbon

```diff
@@ -0,0 +1,30 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
```

Comment:
Perhaps add a few more tests, particularly negative tests?

There's a bunch for generic class tests in `testdata/generic_class`. Probably a few of those could become good choice tests.